sympy/matrices/matrices.py

```diff
@@ -1581,6 +1583,12 @@ def _matrix_pow_by_recursion(self, num):
     def __pow__(self, num):
         if not self.is_square:
             raise NonSquareMatrixError()
+        num = sympify(num)
+        if not num.is_integer and (num >= 0) == True and not num % 1 == 0:
```

Comment:
if num < 0 under the condition when it has any eigenvalue 0 will return ```ValueError("Matrix det == 0; not invertible")```
```not num % 1 == 0``` to exclude float which are integers eg 10.0 as you suggested